common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java

```diff
@@ -352,7 +352,7 @@ public AcquireTokenResult acquireTokenSilent(
 
                 throw exception;
             }
-        } else if (fullCacheRecord.getAccessToken().isExpired()) {
+        } else if (fullCacheRecord.getAccessToken().isExpired() || parameters.isForceRefresh()) {
```

Comment:
I see this fork removes the access token in cache. Not sure if this is intended in case of isForceRefresh == true

Reply:
Yes this should not be needed..because renew may fail and the older AT may still be valid if not expired and we should keep in in cache

Reply:
I think the point is that the code here in the controllers should NOT know what is happening in the Command Dispatcher....when someone reads this code `parameters.isForceRefresh()` - all this should mean is that caller specified that the token should be forced refreshed....it shouldn't matter whether a refresh in was needed or not...

Reply:
@shahzaibj This logic is removed. The PR is using parameters instanceof RefreshInTokenCommandParameters instead.

Reply:
Outdated